### PR TITLE
Catalog Ingestion: response code status checks added

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Added response status checks for bolt requests.

Fixes: (link ticket)

#changelog Catalog Ingestion: response code status checks added

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by garmashdev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1590** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1590" target="_blank">Graphite</a> 
    * **PR #1592** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1592" target="_blank">Graphite</a>  👈
      * **PR #1591** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1591" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1592?utm_source=stack-comment).

File: Model/CatalogIngestion/ProductEventManager.php [hunk 1]
```diff
@@ -196,9 +197,20 @@ public function sendProductEvent(ProductEventInterface $productEvent): bool
...
+                    __(
+                        'Error response status during %1 request, status: %2',
+                        ProductEventRequestBuilder::API_REQUEST_API_URL,
+                        $responseStatus
+                    )
+                );
+            }
         }
-        return true;
+        return false;
```
Comment (vitaliyreznikov) [inline Model/CatalogIngestion/ProductEventManager.php:213]: Do we want to return false in case the feature is disabled for the website?

**Review by vitaliyreznikov — APPROVED** [review]
LGTM, one small question

**Comment by garmashdev** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1592?utm_source=stack-comment).


Job Reference: HiIaPPQEAK5uY3radM0x

**Pull request merged**